### DISCUSSION TITLE
variables

### DISCUSSION THREAD

**Discussion by marco-santulli** [Budibase/budibase#3703] [Ideas]
Hi,

it would be good if we could define variables at the page / app level, so one could put in temp values, it would help a bit in the coding phases,

M

**Comment by marco-santulli** [comment]
bump?

**Comment by cfepdan** [comment]
For what it's worth, I'm going to do this with a Budibase table and a data provider that wraps pretty much my entire app. So in that table, I'll define a set of variables up front that I can access throughout the app.

**Reply by gerhardmpl** [reply]
Did you check the[ app state](https://docs.budibase.com/docs/app-state) values that even can be persisted between page reloads or closing the browser? Changes to the values seem to be bound to button actions, but can be read in JS code nevertheless.

**Reply by cfepdan** [reply]
Yes, but I don't think app state values can be set with an initial value before the page is first loaded, whereas with a table I can preload them.
